mayastor/src/bdev/nexus/nexus_metadata.rs

```diff
@@ -0,0 +1,620 @@
+use std::{
```

Comment:
Add the top of this file, please add a big theory statement where you explain (roughly) how it works and how it can be used. Top file comments like these should be documented using `//!` as a maker.  as an example you see reactor.rs 

(again, all of this renders into nice docs)